### ISSUE TITLE
How to achieve typical organization users table view

### ISSUE THREAD

**Issue by kevinmitch14** [workos/workos-node#1080]
Hey all, 

We are trying to create an organization users view in our application and there doesn't seem to be a straightforward way of doing so. (Image below) Right now to achieve this, I am combining the responses of `listUsers()` and `listOrganizationMembers()`. 

I am basically getting all the users in my WorkOS tenant, regardless of organization_id. Then combining this with the response of listing organization memberships. 

IIRC, when using `listUsers()` and passing the `organization_id`, this only returns the active/accepted users in the org. It doesn't contain pending/inactive users.

This method also has an issue where the max return from `listUsers()` is 100, so if my WorkOS tenant has over 100 users, it becomes a bit messy. (Some users for the given org, will not be returned in the initial 100 results) 

```ts
const { data: users } = await workosClient.userManagement.listUsers({
  limit: 100,
});

const { data: organizationMembers } =
    await workosClient.userManagement.listOrganizationMemberships({
    organizationId,
    statuses: ["active", "inactive", "pending"],
});
```

---

Basically what we are trying to achieve is something like WorkOS's org view. Pretty typical (Name, Email, Picture, Role, Status)
<img width="1072" alt="Screenshot 2024-07-17 at 18 02 17" src="https://github.com/user-attachments/assets/9b6f69f0-83e7-4b38-9674-90c220d7bca3">

**Comment by kevinmitch14** [comment]
Is it safe to say there is no trivial way to accomplish this?

**Comment by PaulAsjes** [comment]
Hey there, you can still accomplish this by using [auto-pagination](https://workos.com/docs/reference/pagination) in your `listUsers` method call, which gets you all your users instead of just the first 100.

However I think the biggest issue here is that you might be using WorkOS as a database to render your organization view. If you're doing the above on a page load you'll start to get longer and longer wait times as your user base grows. We [recommend](https://workos.com/docs/user-management/modeling-your-app/b2b-modeling/user-data-in-b2b-models) you store this data on your end instead with a reference to the WorkOS user ID.

You could rely on [webhooks](https://workos.com/docs/events/role) to keep your database in sync with any user updates instead of having to poll the API every time you want to render this view.

**Comment by kevinmitch14** [comment]
Hey, thanks for the response. What do you mean by 'auto-pagination'? Without stating a limit in `listUsers()` I get 10 results back. 
> Limit: Upper limit on the number of objects to return, between 1 and 100. The default value is 10.

Correct, we are using WorkOS as a 'database' for users/invites etc. We are storing user data in our DB too via webhooks, but WorkOS is the real source of truth, the data can get out of sync if a webhook fails for example. We would like to rely on WorkOS as much as possible for this, right now at least!
